src/test/java/com/hotels/bdp/circustrain/bigquery/client/BigQueryMetastoreClientTest.java

```diff
@@ -0,0 +1,280 @@
+import java.util.List;
+
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.UnknownDBException;
+import org.apache.thrift.TException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobStatus;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableDefinition;
+
+import com.hotels.bdp.circustrain.bigquery.cache.MetastoreClientCache;
+import com.hotels.bdp.circustrain.bigquery.extraction.service.ExtractionService;
+import com.hotels.bdp.circustrain.bigquery.table.service.partitioned.PartitionedTableService;
+import com.hotels.bdp.circustrain.bigquery.table.service.TableServiceFactory;
+import com.hotels.bdp.circustrain.bigquery.util.CircusTrainBigQueryMetastore;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BigQueryMetastoreClientTest {
+
+  private @Mock BigQuery bigQuery;
+  private @Mock ExtractionService extractionService;
+  private @Mock TableServiceFactory factory;
+  private @Mock
+  PartitionedTableService partitionedTableService;
+  private @Mock MetastoreClientCache cache;
+  private @Mock Job job;
+  private @Mock
+  CircusTrainBigQueryMetastore bigQueryMetastore;
+  private @Mock JobStatus jobStatus;
+
+  private BigQueryMetastoreClient bigQueryMetastoreClient;
+
+  @Before
+  public void init() {
+    bigQueryMetastoreClient = new BigQueryMetastoreClient(bigQueryMetastore, extractionService, cache, factory);
+  }
+
+  @Test
+  public void getDatabaseTest() throws TException {
+    when(bigQuery.getDataset(anyString())).thenReturn(mock(Dataset.class));
+    Database database = bigQueryMetastoreClient.getDatabase("test");
+    assertEquals("test", database.getName());
+  }
+
+  @Test(expected = UnknownDBException.class)
+  public void getDatabaseWhenDatabaseDoesntExistThrowsExceptionTest() throws TException {
+    String dbName = "database";
+    doThrow(new UnknownDBException()).when(bigQueryMetastore).checkDbExists(dbName);
+    bigQueryMetastoreClient.getDatabase(dbName);
+  }
+
+  @Test
+  public void tableExistsTest() throws TException {
+    Dataset dataset = mock(Dataset.class);
+    Table table = mock(Table.class);
+    when(bigQuery.getDataset(anyString())).thenReturn(dataset);
+    when(dataset.get(anyString())).thenReturn(table);
+    String dbName = "database";
+    String tblName = "table";
+    bigQueryMetastoreClient.tableExists(dbName, tblName);
+    verify(bigQueryMetastore).tableExists(dbName, tblName);
+  }
+
+  @Test(expected = UnknownDBException.class)
+  public void tableExistsWhenDatabaseDoesntExistThrowsExceptionTest() throws TException {
+    String dbName = "database";
```

Comment:
You use a dbName and tableName in most of the tests so make these member variables so you don't need to repeat them in lots of methods.